### PR TITLE
photon-tx2-nx: Device type added

### PR DESCRIPTION
**Pull request by ClintHaerinck**
Added CTI Photon TX2 NX device type

Changelog-entry: Added CTI Photon TX2 NX device type
Signed-off-by: ClintHaerinck <clint.haerinck@gmail.com>

### CODE REVIEW

**Comment by ClintHaerinck** [comment]
It is linked to issue #207

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 1]
```diff
@@ -66,6 +66,12 @@ SRC_URI_append_photon-nano = " \
     file://tegra210-nano-cti-NGX003.dtb \
 "
 
+SRC_URI_append_photon-tx2-nx = " \
+    file://0001-cti-photon-merge-CDC-MBIM-driver-changes-from-bsp.patch \
```
Comment (acostach) [inline layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:70]: The CDC-MBIM driver changes patch is needed on this carrier board too? If so, we'll keep them.

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 2]
```diff
@@ -187,18 +193,21 @@ BALENA_CONFIGS_append_jetson-nano = " gasket"
 BALENA_CONFIGS_append_jetson-nano-emmc = " gasket"
 
 BALENA_CONFIGS_append_photon-nano = " tlc591xx"
+BALENA_CONFIGS_append_photon-tx2-nx = " tlc591xx"
```
Comment (acostach) [inline layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:196]: I assume this is is necessary for the photon tx2 nx too, correct?

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 3]
```diff
@@ -219,6 +229,7 @@ BALENA_CONFIGS_DEPS[cdc-ncm] = " \
 "
 
 BALENA_CONFIGS_append_photon-nano = " mii"
+BALENA_CONFIGS_append_photon-tx2-nx = " mii"
```
Comment (acostach) [inline layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:232]: same for this and the other configs above?

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend [hunk 4]
```diff
@@ -347,6 +358,11 @@ do_deploy_append_photon-nano() {
     cp ${WORKDIR}/tegra210-nano-cti-NGX003.dtb "${DEPLOYDIR}"
 }
 
+do_deploy_append_photon-tx2-nx() {
+    cp ${WORKDIR}/tegra186-tx2-nx-cti-NGX003.dtb "${DEPLOYDIR}"
+    cp ${WORKDIR}/tegra186-tx2-nx-cti-NGX003-IMX219-2CAM.dtb "${DEPLOYDIR}"
+}
```
Comment (acostach) [inline layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:364]: Let's ship these dtbs with the jetson-dtbs package so they are available in the rootfs boot directory: https://github.com/balena-os/balena-jetson/blob/master/layers/meta-balena-jetson/recipes-kernel/linux/jetson-dtbs.bb

**Review by acostach — COMMENTED** [review]
Thanks for the PR, it looks ok. I left a couple notes to clarify whether some configs are necessary on this carrier board and we'll also need to include the Photon TX2 NX dtbs in the rootfs boot directory. Were you able to to a local build and test this image? Did it work fine?

**Comment by acostach** [comment]
@resin-jenkins test this please

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend (hunk 4 above)
Comment (ClintHaerinck) [reply layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:364]: Ok sounds good. I looked at the photon-nano as example and they aren't included for that device.

**Comment by ClintHaerinck** [comment]
> Thanks for the PR, it looks ok. I left a couple notes to clarify whether some configs are necessary on this carrier board and we'll also need to include the Photon TX2 NX dtbs in the rootfs boot directory. Were you able to to a local build and test this image? Did it work fine?

I've made a local build successful but still have to test the image. I let you know when tested.

File: layers/meta-balena-jetson/conf/machine/photon-tx2-nx.conf [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#@TYPE: Machine
+#@NAME: CTI Photon TX2 NX
+##@DESCRIPTION: CTI Photon Carrier Board for NVIDIA Jetson TX2 NX
+#
+
+MACHINEOVERRIDES = "jetson-tx2-nx-devkit:${MACHINE}"
+include conf/machine/jetson-tx2-nx-devkit.conf
+
+# work-around for https://github.com/OE4T/meta-tegra/issues/400
+MACHINEOVERRIDES := "${@'${MACHINEOVERRIDES}'.replace(':${MACHINE}',':jetson-tx2-nx-devkit:${MACHINE}')}"
```
Comment (pgils) [inline layers/meta-balena-jetson/conf/machine/photon-tx2-nx.conf:10]: Hi @ClintHaerinck 
This override shouldn't be needed anymore. It was fixed in https://github.com/OE4T/meta-tegra/commit/46b020ef63550bd5c6534319824082347e25b58b

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend (hunk 1 above)
Comment (ClintHaerinck) [reply layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:70]: I think so, just validated a modem is working

File: layers/meta-balena-jetson/conf/machine/photon-tx2-nx.conf (hunk 5 above)
Comment (ClintHaerinck) [reply layers/meta-balena-jetson/conf/machine/photon-tx2-nx.conf:10]: Thanks, I will commit the change

**Comment by ClintHaerinck** [comment]
I've validated the build, it's working. However, I can't connect to BalenaCloud with this device because of the new device type.

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend (hunk 3 above)
Comment (ClintHaerinck) [reply layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:232]: Ethernet is working

**Comment by acostach** [comment]
@ClintHaerinck it needs to become available in the API first to show up online. For that, the PR needs to be updated and merged.

File: layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend (hunk 2 above)
Comment (ClintHaerinck) [reply layers/meta-balena-jetson/recipes-kernel/linux/linux-tegra_%.bbappend:196]: Yes

File: layers/meta-balena-jetson/conf/machine/photon-tx2-nx.conf (hunk 5 above)
Comment (ClintHaerinck) [reply layers/meta-balena-jetson/conf/machine/photon-tx2-nx.conf:10]: Deleting the 2 lines doesn't work. More investigation is necessary.

**Comment by ClintHaerinck** [comment]
Please merge the device type. I've tested the build, it works.

**Comment by acostach** [comment]
@ClintHaerinck can you please remove the two commits that remove and back the machine overrides? Also, one Changelog-entry per PR is sufficient, and Signed-off-by is needed on every commit (already done, OK). Thanks

**Comment by ClintHaerinck** [comment]
> @ClintHaerinck can you please remove the two commits that remove and back the machine overrides? Also, one Changelog-entry per PR is sufficient, and Signed-off-by is needed on every commit (already done, OK). Thanks

Done

**Comment by ClintHaerinck** [comment]
@acostach There was a check failing. Now should everything be OK.

**Review by acostach — APPROVED** [review]

**Pull request merged**